WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsServiceLogic.java

```diff
@@ -0,0 +1,627 @@
+                            maxResultsRequested, pageRequested, (CommentsModel) data);
+                case COMMENT_FOLLOWERS:
+                    return new StatsEvents.CommentFollowersUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (CommentFollowersModel) data);
+                case TAGS_AND_CATEGORIES:
+                    return new StatsEvents.TagsUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested, (TagsContainerModel) data);
+                case PUBLICIZE:
+                    return new StatsEvents.PublicizeUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested, (PublicizeModel) data);
+                case FOLLOWERS_WPCOM:
+                    return new StatsEvents.FollowersWPCOMUdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (FollowersModel) data);
+                case FOLLOWERS_EMAIL:
+                    return new StatsEvents.FollowersEmailUdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (FollowersModel) data);
+                case INSIGHTS_POPULAR:
+                    return new StatsEvents.InsightsPopularUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (InsightsPopularModel) data);
+                case INSIGHTS_ALL_TIME:
+                    return new StatsEvents.InsightsAllTimeUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (InsightsAllTimeModel) data);
+                case INSIGHTS_TODAY:
+                    return new StatsEvents.VisitorsAndViewsUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (VisitsModel) data);
+                case INSIGHTS_LATEST_POST_SUMMARY:
+                    return new StatsEvents.InsightsLatestPostSummaryUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (InsightsLatestPostModel) data);
+                case INSIGHTS_LATEST_POST_VIEWS:
+                    return new StatsEvents.InsightsLatestPostDetailsUpdated(siteId, timeframe, date,
+                            maxResultsRequested, pageRequested,
+                            (InsightsLatestPostDetailsModel) data);
+                default:
+                    AppLog.e(T.STATS, "Can't find an Update Event that match the current endpoint: " + this.name());
+            }
+
+            return null;
+        }
+    }
+
+    public StatsServiceLogic(ServiceCompletionListener completionListener) {
+        mCompletionListener = completionListener;
+    }
+
+    public void onCreate(WordPress app) {
```

Comment:
I think `onCreate` and `onDestroy` can be removed here.

Reply:
Oh no these are attached to both `StatsJobService` and `StatsService` lifecycle implementations, like this for example in StatsService:
```
    @Override
    public void onCreate() {
        super.onCreate();
        AppLog.i(T.STATS, "stats service created");
        mStatsServiceLogic = new StatsServiceLogic(this);
        mStatsServiceLogic.onCreate((WordPress) getApplication());
    }

    @Override
    public void onDestroy() {
        AppLog.i(T.STATS, "stats service destroyed");
        mStatsServiceLogic.onDestroy();
        super.onDestroy();
    }
```

I could also enforce this by writing a simple `interface` to make it more clear that these need to exist if you want.

Reply:
Ah sorry, thought it was a leftover from the previous class.